### PR TITLE
Update login state management

### LINKED ISSUE #181: Login: Update state management of login
### Describe the feature

This issue aims to refactor the management of the jwt token and session on the frontend side of the application.

### Proposed Solution

Using the [zustand](https://zustand-demo.pmnd.rs/) library create a global state to manage the session on the browser and using the [TypeScript Cookie](https://github.com/carhartl/typescript-cookie#readme) create a Cookie to save the jwt token.

### PR DESCRIPTION
**Pull request by fungame2270**
### Issue

Closes #181 .

### Reason for this change

Fixes some problems like redirect and password showing

### Description of changes

changed how the navbar checks login created new component for protected route

### Description of how you validated changes

tested locally

### Checklist
- [X] I have tested my changes locally.
- [ ] I have updated the documentation to reflect my changes.
- [ ] I have added/updated unit tests (if applicable).
- [X] My code adheres to the [CONTRIBUTING GUIDE](https://ua-smart-signage-platform.github.io/Documentation-Website/guidelines/)

### CODE REVIEW

**Comment by pdPinho** [comment]
There are some code smells that require fixing

File: frontend/src/routes/Root/index.jsx [hunk 1]
```diff
@@ -30,9 +30,18 @@ const queryClient = new QueryClient();
...
     const location = useLocation();
 
+    useEffect(()=>{
+      if (!logged){
+        navigate("/login");
+      }else if (location.pathname === "/"){
+        navigate("/dashboard")
+      }
+    },[logged])
+
```
Comment (pdPinho) [inline frontend/src/routes/Root/index.jsx]: if the role isn't admin, what happens then?

**Review by pdPinho — CHANGES_REQUESTED** [review]

File: frontend/src/routes/Root/index.jsx (hunk 1 above)
Comment (fungame2270) [reply frontend/src/routes/Root/index.jsx:44]: heheha

Comment (fungame2270) [reply frontend/src/routes/Root/index.jsx]: pagina errada

**Review by pdPinho — APPROVED** [review]
ok

**Pull request merged**